[4794][IMP] account_commission_sales_fee

AungKoKoLin1997 opened the pull request:
[4794](https://www.quartile.co/web#id=4794&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

AungKoKoLin1997 committed:
[IMP] account_commission_sales_fee

yostashiro commented on account_commission_sales_fee/models/commission_settlement.py:
```diff
@@ -11,7 +11,10 @@ def _prepare_sales_fee_product(self, commission):
             "product_id": commission.sales_fee_product_id.id,
             "quantity": len(
-                self.line_ids.filtered(lambda line: line.commission_id == commission)
+                self.line_ids.filtered(
+                    lambda line: line.commission_id == commission
+                    and line.invoice_line_id.move_id.move_type == "out_invoice"
+                )
```
I don't think we should rely on move_type as out_invoice could also hold lines with negative subtotals.
```suggestion
                self.line_ids.filtered(
                    lambda line: line.commission_id == commission and line.settled_amount > 0.0
                )
```

AungKoKoLin1997 replied:
I updated the code.

yostashiro reviewed (approved):
Code review.

yostashiro reviewed (changes requested):
Sorry, I approved too early.

@AungKoKoLin1997 Can you please sort out the test error?

AungKoKoLin1997 committed:
[FIX] account_commission_sales_fee

This commit updates the commission type from 'section' to 'net_amount.' If we use the 'section' type commission and
the commission amount is outside the section range, the commission value returns as 0.0. This commit addresses this issue.

yostashiro reviewed (commented):
@AungKoKoLin1997 Is the test fix related to the update of the code this time? I couldn't tell.

AungKoKoLin1997 commented:
> @AungKoKoLin1997 Is the test fix related to the update of the code this time? I couldn't tell.

Yes. It is. Now we added `settled_amount` in filtered. With pervious code, the settlement lines is 2 lines but `settled_amount` are `0.0` because the `amount` in the agent line of invoice line is 0.0. It is because of the base amount is not within the range of section_ids. https://github.com/qrtl/rmm-oca/blob/b88195713c3f92f3a30364de5d374c59ef94eae0/commission/models/commission.py#L40-L45.
So, instead of checking how section_ids was defined and use the amount within this range, I decided to change commission type.

yostashiro commented:
@AungKoKoLin1997 Thanks Lin. I think the tests could be improved but it's OK for now.

kanda999 reviewed (approved):
Functional Check
